File: eSalon/eSalon.Model/AktiviranaPromocija.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace eSalon.Model
{
    public class AktiviranaPromocija
    {
        public int AktiviranaPromocijaId { get; set; }

        public int PromocijaId { get; set; }

        public int KorisnikId { get; set; }

        public bool Aktivirana { get; set; }

        public bool Iskoristena { get; set; }

        public DateTime DatumAktiviranja { get; set; }
        public string PromocijaNaziv { get; set; } = string.Empty;
        public string KorisnikImePrezime { get; set; } = string.Empty;

        public string? KodPromocije { get; set; }
        public byte[]? SlikaUsluge { get; set; }

        public decimal? Popust { get; set; }

        public DateTime? DatumPocetka { get; set; }

        public DateTime? DatumKraja { get; set; }

    }
}

File: eSalon/UI/esalon_mobile/lib/models/aktivirana_promocija.dart
import 'package:json_annotation/json_annotation.dart';

part 'aktivirana_promocija.g.dart';

@JsonSerializable()
class AktiviranaPromocija {
  int? aktiviranaPromocijaId;
  int? promocijaId;
  int? korisnikId;
  bool? aktivirana;
  bool? iskoristena;
  DateTime? datumAktiviranja;
  String? promocijaNaziv;
  String? korisnikImePrezime;
  String? kodPromocije;
  String? slikaUsluge; 
  double? popust;
  DateTime? datumPocetka;
  DateTime? datumKraja;

  AktiviranaPromocija({
    this.aktiviranaPromocijaId,
    this.promocijaId,
    this.korisnikId,
    this.aktivirana,
    this.iskoristena,
    this.datumAktiviranja,
    this.promocijaNaziv,
    this.korisnikImePrezime,
    this.kodPromocije,
    this.slikaUsluge,
    this.popust,
    this.datumPocetka,
    this.datumKraja,
  });

  factory AktiviranaPromocija.fromJson(Map<String, dynamic> json) =>
      _$AktiviranaPromocijaFromJson(json);

  Map<String, dynamic> toJson() => _$AktiviranaPromocijaToJson(this);
}
